controllers/util/mw_util.go

```diff
@@ -412,22 +412,37 @@ func (mwu *MWUtil) deleteManifestWork(mwName, mwNamespace string) error {
 	return nil
 }
 
-func GetMetricValueSingle(name string, mfType dto.MetricType) (float64, error) {
-	mf, err := getMetricFamilyFromRegistry(name)
+func GetMetricValueSingle(name string, mfType dto.MetricType, registry metrics.RegistererGatherer) (float64, error) {
```

Comment:
That's correct. The main code only really needs to call the updates for each of the metrics, but the `GetMetricValueSingle` (etc) functions needed validation, so that's why those exist. Referencing the values of the metrics at runtime happens through an external curl request or a dashboard, rather than inside the code somewhere.